src/core_plugins/kbn_vislib_vis_types/public/controls/heatmap_options.js

```diff
@@ -0,0 +1,56 @@
+import uiModules from 'ui/modules';
+import heatmapOptionsTemplate from 'plugins/kbn_vislib_vis_types/controls/heatmap_options.html';
+import _ from 'lodash';
+const module = uiModules.get('kibana');
+
+module.directive('heatmapOptions', function ($parse, $compile, getAppState) {
+  return {
+    restrict: 'E',
+    template: heatmapOptionsTemplate,
+    replace: true,
+    link: function ($scope) {
+      $scope.isColorRangeOpen = true;
+      $scope.customColors = false;
+      $scope.options = {
+        labels: false
+      };
+
+      $scope.$watch('options.labels', rotate => {
+        $scope.vis.params.valueAxes[0].labels.rotate = rotate ? 270 : 0;
+      });
+
+      $scope.resetColors = () => {
+        $scope.uiState.set('vis.colors', null);
+        $scope.customColors = false;
+      };
+
+      $scope.getGreaterThan = function (index) {
```

Comment:
+1. 

the below is a little counter-intuitive:
![image](https://cloud.githubusercontent.com/assets/1833023/21481204/f035daa0-cb32-11e6-807d-2a119b0dba35.png)